ui/store/actions.ts

```diff
@@ -2989,6 +2990,35 @@ export function setPreference(
             reject(err);
             return;
           }
+          console.log('updatedPreferences', updatedPreferences);
+          resolve(updatedPreferences as TemporaryPreferenceFlagDef);
+        },
+      );
+    });
+  };
+}
+
+export function setPreferenceWithoutLoader(
```

Comment:
Might be simpler to add a `showLoading` flag to the existing `setPreference`, and have it default to true

Reply:
https://github.com/MetaMask/metamask-extension/pull/27184/commits/6fc7bfc7d08b67fc61887d72d30719dec8439646